src/connections.c

```diff
@@ -1071,7 +1071,7 @@ connection_revents_err (request_st * const r, connection * const con)
           ~(FDEVENT_STREAM_REQUEST_BUFMIN|FDEVENT_STREAM_REQUEST_POLLIN);
         r->conf.stream_request_body |= FDEVENT_STREAM_REQUEST_POLLRDHUP;
         con->is_readable = 1; /*(can read 0 for end-of-stream)*/
-        if (chunkqueue_is_empty(con->read_queue)) r->keep_alive = 0;
```

Comment:
This is wrong.  If there is data that has been read from the client socket, but not yet processed, lighttpd attempts to process that data before considering the read side of the connection to have been shutdown (SHUT_RD)

Reply:
No.  Wrong again.  `git log -G 'r->keep_alive'`